cdc/processor/sourcemanager/sorter/pebble/event_sorter.go

```diff
@@ -463,21 +464,36 @@ func (s *EventSorter) handleEvents(
 		}
 	}
 
-	for {
-		for len(batch.Repr()) < batchCommitSize && time.Since(startToCollectBatch) < batchCommitInterval {
+	// Batch item and commit until batch size is larger than batchCommitSize,
+	// or the time since the last commit is larger than batchCommitInterval.
+	// Only return false when the sorter is closed.
+	doBatching := func() (*DBBatchEvent, bool) {
+		batch := db.NewBatch()
+		newResolved := spanz.NewHashMap[model.Ts]()
+		startToBatch := time.Now()
+		for {
 			select {
 			case item := <-inputCh:
-				handleItem(item)
+				encodeItemAndBatch(batch, newResolved, item)
+				if len(batch.Repr()) >= batchCommitSize {
+					return &DBBatchEvent{batch, newResolved}, true
+				}
 			case <-s.closed:
-				return
-			default:
+				return nil, false
+			case <-ticker.C:
+				if time.Since(startToBatch) >= batchCommitInterval {
+					return &DBBatchEvent{batch, newResolved}, true
+				}
 			}
 		}
-		batchCh <- &DBBatchEvent{batch, newResolved}
+	}
 
-		batch = db.NewBatch()
-		newResolved = spanz.NewHashMap[model.Ts]()
-		startToCollectBatch = time.Now()
+	for {
+		batchEvent, ok := doBatching()
+		if !ok {
+			return
+		}
+		batchCh <- batchEvent
```

Comment:
should an empty batchEvent be ignored?

Reply:
I check whether batchEvent is empty in L395, because whether the batchEvent is empty, we have to  update resolved ts.